Gymphony/src/app/features/courses/pages/course-schedules/course-schedules.component.ts

```diff
@@ -0,0 +1,120 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { loadStripe } from '@stripe/stripe-js';
+import { catchError, EMPTY, filter, finalize, switchMap, tap } from 'rxjs';
+
+import { MessageService } from '../../../../shared/services/message.service';
+import { CoursesService } from '../../services/courses.service';
+import { CourseSchedule } from '../../interfaces/course-schedule';
+import { SubscriptionsService } from '../../../user-profile/services/subscriptions.service';
+import { CheckoutSession } from '../../../membership-plans/interfaces/checkout-session';
+import { SubscribeForCourse } from '../../interfaces/subscribe-for-course';
+import { ApiError } from '../../../../core/interfaces/api-error';
+import { Course } from '../../interfaces/course';
+import { PublicCourses } from '../../interfaces/public-courses';
+import { LoaderService } from '../../../../core/services/loader.service';
+
+@Component({
+  selector: 'app-course-schedules',
+  templateUrl: './course-schedules.component.html',
+  styleUrl: './course-schedules.component.css'
+})
+export class CourseSchedulesComponent implements OnInit {
+  public courseId!: string;
+  public course!: Course;
+  public sessionsCount!: number;
+  public courseSchedules: CourseSchedule[] = [];
+  public selectedScheduleIds: string[] = [];
+  public courses!: Course[];
+  public selectedCourseName!: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private messageService: MessageService,
+    private coursesService: CoursesService,
+    private subscriptionsService: SubscriptionsService,
+    private router: Router,
+    private loaderService: LoaderService) { }
+
+  public ngOnInit(): void {
+    this.route.queryParams.pipe(
+      filter(params => !!params['courseId']),
+      tap(params => {
+        this.courseId = params['courseId'];
+        this.sessionsCount = params['sessions'];
+      }),
+      switchMap(() => this.coursesService.getCourseById(this.courseId)),
+      tap((course: Course) => {
+        this.course = course;
+        this.selectedCourseName = course.name;
+      }),
+      switchMap(() => this.coursesService.getActiveCourseSchedules(this.courseId)),
+      tap((courseSchedules: CourseSchedule[]) => {
+        this.courseSchedules = courseSchedules;
+      }),
+      switchMap(() => this.coursesService.getPublicCourses()),
```

Comment:
If you need multiple http requests at one time, you could use forkJoin operator, don't create such long sequence of switchMaps